api/tasks.go

```diff
@@ -1,20 +1,18 @@
 package api
 
 import (
-	"context"
 	"encoding/json"
-	"net/http"
-
 	"github.com/ONSdigital/dp-search-reindex-api/models"
 	"github.com/ONSdigital/dp-search-reindex-api/mongo"
 	"github.com/ONSdigital/log.go/v2/log"
 	"github.com/gorilla/mux"
+	"net/http"
 )
 
 var invalidBodyErrorMessage = "invalid request body"
 
 // CreateTaskHandler returns a function that generates a new TaskName resource containing default values in its fields.
-func (api *API) CreateTaskHandler(ctx context.Context) http.HandlerFunc {
+func (api *API) CreateTaskHandler(taskNames map[string]bool) http.HandlerFunc {
```

Comment:
If you added taskNames into the api struct you wouldn't need to create a function-that-returns-a-function and pass it as an argument.. Would make this section of the code a little easier to read (and is generally a better pattern to use IMO)

Reply:
Good idea, I prefer that pattern too now you've pointed it out, thanks.